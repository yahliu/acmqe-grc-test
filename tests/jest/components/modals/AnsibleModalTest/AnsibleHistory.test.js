/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import { renderAnsibleHistory } from '../../../../../src-web/components/modals/AnsibleAutomationModal/AnsibleHistory'

const mockAnsibleJobListResponse1 = {
  apiVersion: 'tower.ansible.com/v1alpha1',
  items: [],
  kind: 'AnsibleJobList',
  metadata: {
    continue: '',
    resourceVersion: '376760',
    selfLink: '/apis/tower.ansible.com/v1alpha1/namespaces/default/ansiblejobs',
  },
}

const mockAnsibleJobListResponse2 = {
  apiVersion: 'tower.ansible.com/v1alpha1',
  items: [
    {
      apiVersion: 'tower.ansible.com/v1alpha1',
      kind: 'AnsibleJob',
      metadata: {
        creationTimestamp: '2021-05-03T17:18:10Z',
        generateName: 'create-service-now-ticket-once-',
        generation: 1,
        labels: {
          tower_job_id: '20',
        },
        name: 'create-service-now-ticket-once-g6xsr',
        namespace: 'default',
        ownerReferences: [
          {
            apiVersion: 'policy.open-cluster-management.io/v1beta1',
            blockOwnerDeletion: true,
            controller: true,
            kind: 'PolicyAutomation',
            name: 'create-service-now-ticket',
            uid: '',
          },
        ],
        resourceVersion: '358239',
        selfLink: '/apis/tower.ansible.com/v1alpha1/namespaces/default/ansiblejobs/create-service-now-ticket-once-g6xsr',
        uid: '',
      },
      spec: {
        extra_vars: {
          sn_priority: 1,
          sn_severity: 1,
          target_clusters: [
            'local-cluster',
          ],
        },
        job_template_name: 'Demo Job Template',
      },
      status: {
        ansibleJobResult: {
          changed: true,
          elapsed: '6.384',
          failed: false,
          finished: '2021-05-03T17:18:43.332157Z',
          started: '2021-05-03T17:18:36.948508Z',
          status: 'successful',
          url: 'https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-z4bg2.dev08.red-chesterfield.com/#/jobs/playbook/20',
        },
        conditions: [
          {
            ansibleResult: {
              changed: 0,
              completion: '2021-05-03T17:18:55.54961',
              failures: 0,
              ok: 3,
              skipped: 0,
            },
            lastTransitionTime: '2021-05-03T17:18:10Z',
            message: 'Awaiting next reconciliation',
            reason: 'Successful',
            status: 'True',
            type: 'Running',
          },
        ],
        k8sJob: {
          created: true,
          env: {
            templateName: 'Demo Job Template',
            verifySSL: false,
          },
          message: 'Monitor the job.batch status for more details with the following commands:\n\'kubectl -n default get job.batch/create-service-now-ticket-once-g6xsr\'\n\'kubectl -n default describe job.batch/create-service-now-ticket-once-g6xsr\'\n\'kubectl -n default logs -f job.batch/create-service-now-ticket-once-g6xsr\'',
          namespacedName: 'default/create-service-now-ticket-once-g6xsr',
        },
        message: 'This job instance is already running or has reached its end state.',
      },
    },
    {
      apiVersion: 'tower.ansible.com/v1alpha1',
      kind: 'AnsibleJob',
      metadata: {
        creationTimestamp: '2021-05-03T17:31:32Z',
        generateName: 'policy-role-once-',
        generation: 1,
        labels: {
          tower_job_id: '22',
        },
        name: 'policy-role-once-7fg9p',
        namespace: 'default',
        ownerReferences: [
          {
            apiVersion: 'policy.open-cluster-management.io/v1beta1',
            blockOwnerDeletion: true,
            controller: true,
            kind: 'PolicyAutomation',
            name: 'policy-role',
            uid: null,
          },
        ],
        resourceVersion: '371219',
        selfLink: '/apis/tower.ansible.com/v1alpha1/namespaces/default/ansiblejobs/policy-role-once-7fg9p',
        uid: null,
      },
      spec: {
        extra_vars: {
          sn_priority: 1,
          sn_severity: 1,
          target_clusters: [
            'local-cluster',
          ],
        },
        job_template_name: 'Demo Job Template',
      },
      status: {
        ansibleJobResult: {
          changed: true,
          elapsed: '6.578',
          failed: false,
          finished: '2021-05-03T17:32:06.357295Z',
          started: '2021-05-03T17:31:59.779211Z',
          status: 'successful',
          url: 'https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-z4bg2.dev08.red-chesterfield.com/#/jobs/playbook/22',
        },
        conditions: [
          {
            ansibleResult: {
              changed: 0,
              completion: '2021-05-03T17:32:18.620627',
              failures: 0,
              ok: 3,
              skipped: 0,
            },
            lastTransitionTime: '2021-05-03T17:31:32Z',
            message: 'Awaiting next reconciliation',
            reason: 'Successful',
            status: 'True',
            type: 'Running',
          },
        ],
        k8sJob: {
          created: true,
          env: {
            templateName: 'Demo Job Template',
            verifySSL: false,
          },
          message: 'Monitor the job.batch status for more details with the following commands:\n\'kubectl -n default get job.batch/policy-role-once-7fg9p\'\n\'kubectl -n default describe job.batch/policy-role-once-7fg9p\'\n\'kubectl -n default logs -f job.batch/policy-role-once-7fg9p\'',
          namespacedName: 'default/policy-role-once-7fg9p',
        },
        message: 'This job instance is already running or has reached its end state.',
      },
    },
    {
      apiVersion: 'tower.ansible.com/v1alpha1',
      kind: 'AnsibleJob',
      metadata: {
        creationTimestamp: '2021-05-03T17:31:32Z',
        generateName: 'policy-role-once-',
        generation: 1,
        labels: {
          tower_job_id: '22',
        },
        name: 'policy-role-once-7fg9p',
        namespace: 'default',
        ownerReferences: [
          {
            apiVersion: 'policy.open-cluster-management.io/v1beta1',
            blockOwnerDeletion: true,
            controller: true,
            kind: 'PolicyAutomation',
            name: 'policy-role',
            uid: 'bddcd0d3-341e-4480-b2ed-e62a428f4ee5',
          },
        ],
        resourceVersion: '371219',
        selfLink: '/apis/tower.ansible.com/v1alpha1/namespaces/default/ansiblejobs/policy-role-once-7fg9p',
        uid: 'c61d3d3c-a8b8-48ed-9afb-b94ab0cc447b',
      },
      spec: {
        extra_vars: {
          sn_priority: 1,
          sn_severity: 1,
          target_clusters: [
            'local-cluster',
          ],
        },
        job_template_name: 'Demo Job Template',
      },
      status: {
        conditions: [
          {
            ansibleResult: {
              changed: 0,
              completion: '2021-05-03T17:32:18.620627',
              failures: 0,
              ok: 3,
              skipped: 0,
            },
            lastTransitionTime: '2021-05-03T17:31:32Z',
            message: 'Tower Secret must exists',
            reason: 'Failed',
            status: 'True',
            type: 'Failure',
          },
        ],
      },
    },
  ],
  kind: 'AnsibleJobList',
  metadata: {
    continue: '',
    resourceVersion: '376760',
    selfLink: '/apis/tower.ansible.com/v1alpha1/namespaces/default/ansiblejobs',
  },
}

describe('render AnsibleHistory component', () => {
  it('render empty AnsibleHistory page', () => {
    const component = renderAnsibleHistory(mockAnsibleJobListResponse1, 'us')
    expect(component).toMatchSnapshot()
  })

  it('render AnsibleHistory page', () => {
    const component = renderAnsibleHistory(mockAnsibleJobListResponse2, 'us')
    expect(component).toMatchSnapshot()
  })
})
