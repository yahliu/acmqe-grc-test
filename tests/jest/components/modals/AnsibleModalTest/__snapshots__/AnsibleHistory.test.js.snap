// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AnsibleHistory component render AnsibleHistory page 1`] = `
<AcmTable
  autoHidePagination={true}
  columns={
    Array [
      Object {
        "cell": "status",
        "cellTransforms": Array [
          [Function],
        ],
        "header": "Status",
        "search": [Function],
        "sort": "status",
        "transforms": undefined,
      },
      Object {
        "cell": "started",
        "cellTransforms": undefined,
        "header": "Started",
        "search": undefined,
        "sort": "started",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "cell": "finished",
        "cellTransforms": undefined,
        "header": "Finished",
        "search": undefined,
        "sort": "finished",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "cell": "link",
        "cellTransforms": undefined,
        "header": "",
        "search": undefined,
        "sort": undefined,
        "transforms": undefined,
      },
    ]
  }
  gridBreakPoint=""
  initialSort={
    Object {
      "direction": "desc",
      "index": 1,
    }
  }
  items={
    Array [
      Object {
        "finished": Object {
          "rawData": undefined,
          "title": <TableTimestamp />,
        },
        "link": Object {
          "rawData": undefined,
          "title": "",
        },
        "started": Object {
          "rawData": undefined,
          "title": <TableTimestamp />,
        },
        "status": Object {
          "rawData": Object {
            "ansibleJobResult": Object {
              "changed": true,
              "elapsed": "6.384",
              "failed": false,
              "finished": "2021-05-03T17:18:43.332157Z",
              "started": "2021-05-03T17:18:36.948508Z",
              "status": "successful",
              "url": "https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-z4bg2.dev08.red-chesterfield.com/#/jobs/playbook/20",
            },
            "conditions": Array [
              Object {
                "ansibleResult": Object {
                  "changed": 0,
                  "completion": "2021-05-03T17:18:55.54961",
                  "failures": 0,
                  "ok": 3,
                  "skipped": 0,
                },
                "lastTransitionTime": "2021-05-03T17:18:10Z",
                "message": "Awaiting next reconciliation",
                "reason": "Successful",
                "status": "True",
                "type": "Running",
              },
            ],
            "k8sJob": Object {
              "created": true,
              "env": Object {
                "templateName": "Demo Job Template",
                "verifySSL": false,
              },
              "message": "Monitor the job.batch status for more details with the following commands:
'kubectl -n default get job.batch/create-service-now-ticket-once-g6xsr'
'kubectl -n default describe job.batch/create-service-now-ticket-once-g6xsr'
'kubectl -n default logs -f job.batch/create-service-now-ticket-once-g6xsr'",
              "namespacedName": "default/create-service-now-ticket-once-g6xsr",
            },
            "message": "This job instance is already running or has reached its end state.",
          },
          "title": <div>
            <YellowExclamationTriangleIcon />
             
            No status
          </div>,
        },
        "uid": 0,
      },
      Object {
        "finished": Object {
          "rawData": undefined,
          "title": <TableTimestamp />,
        },
        "link": Object {
          "rawData": undefined,
          "title": "",
        },
        "started": Object {
          "rawData": undefined,
          "title": <TableTimestamp />,
        },
        "status": Object {
          "rawData": Object {
            "ansibleJobResult": Object {
              "changed": true,
              "elapsed": "6.578",
              "failed": false,
              "finished": "2021-05-03T17:32:06.357295Z",
              "started": "2021-05-03T17:31:59.779211Z",
              "status": "successful",
              "url": "https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-z4bg2.dev08.red-chesterfield.com/#/jobs/playbook/22",
            },
            "conditions": Array [
              Object {
                "ansibleResult": Object {
                  "changed": 0,
                  "completion": "2021-05-03T17:32:18.620627",
                  "failures": 0,
                  "ok": 3,
                  "skipped": 0,
                },
                "lastTransitionTime": "2021-05-03T17:31:32Z",
                "message": "Awaiting next reconciliation",
                "reason": "Successful",
                "status": "True",
                "type": "Running",
              },
            ],
            "k8sJob": Object {
              "created": true,
              "env": Object {
                "templateName": "Demo Job Template",
                "verifySSL": false,
              },
              "message": "Monitor the job.batch status for more details with the following commands:
'kubectl -n default get job.batch/policy-role-once-7fg9p'
'kubectl -n default describe job.batch/policy-role-once-7fg9p'
'kubectl -n default logs -f job.batch/policy-role-once-7fg9p'",
              "namespacedName": "default/policy-role-once-7fg9p",
            },
            "message": "This job instance is already running or has reached its end state.",
          },
          "title": <div>
            <YellowExclamationTriangleIcon />
             
            No status
          </div>,
        },
        "uid": 1,
      },
      Object {
        "finished": Object {
          "rawData": undefined,
          "title": <TableTimestamp />,
        },
        "link": Object {
          "rawData": undefined,
          "title": "",
        },
        "started": Object {
          "rawData": undefined,
          "title": <TableTimestamp />,
        },
        "status": Object {
          "rawData": Object {
            "conditions": Array [
              Object {
                "ansibleResult": Object {
                  "changed": 0,
                  "completion": "2021-05-03T17:32:18.620627",
                  "failures": 0,
                  "ok": 3,
                  "skipped": 0,
                },
                "lastTransitionTime": "2021-05-03T17:31:32Z",
                "message": "Tower Secret must exists",
                "reason": "Failed",
                "status": "True",
                "type": "Failure",
              },
            ],
          },
          "title": <div>
            <YellowExclamationTriangleIcon />
             
            No status
          </div>,
        },
        "uid": 2,
      },
    ]
  }
  keyFn={[Function]}
  plural="ansible jobs"
  showToolbar={false}
/>
`;

exports[`render AnsibleHistory component render empty AnsibleHistory page 1`] = `
<AcmTable
  autoHidePagination={true}
  columns={
    Array [
      Object {
        "cell": "status",
        "cellTransforms": Array [
          [Function],
        ],
        "header": "Status",
        "search": [Function],
        "sort": "status",
        "transforms": undefined,
      },
      Object {
        "cell": "started",
        "cellTransforms": undefined,
        "header": "Started",
        "search": undefined,
        "sort": "started",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "cell": "finished",
        "cellTransforms": undefined,
        "header": "Finished",
        "search": undefined,
        "sort": "finished",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "cell": "link",
        "cellTransforms": undefined,
        "header": "",
        "search": undefined,
        "sort": undefined,
        "transforms": undefined,
      },
    ]
  }
  gridBreakPoint=""
  initialSort={
    Object {
      "direction": "desc",
      "index": 1,
    }
  }
  items={Array []}
  keyFn={[Function]}
  plural="ansible jobs"
  showToolbar={false}
/>
`;
